pkg/querier/blocks_store_queryable.go

```diff
@@ -261,7 +264,12 @@ func NewBlocksStoreQueryableFromConfig(querierCfg Config, gatewayCfg storegatewa
 		reg,
 	)
 
-	return NewBlocksStoreQueryable(stores, finder, consistency, limits, querierCfg.QueryStoreAfter, logger, reg)
+	streamingBufferSize := querierCfg.StreamingChunksPerIngesterSeriesBufferSize
+	if !querierCfg.PreferStreamingChunks {
+		streamingBufferSize = 0
+	}
```

Comment:
What would be the added benefit of having 2 flags? In what case will we configure different batch sizes? Unless we know we need different, I would avoid adding a new flag prematurely.

Reply:
There are two flags here:

* one to enable streaming 
* one to set the batch size when streaming

For the first, it would be useful to be able to enable / disable streaming from store-gateways independently of ingesters, especially while we're testing the feature.

For the second: we usually run with dramatically different numbers of ingesters and store-gateways, and ingesters will generally return far fewer chunks per series for a query (and therefore a single series will consume much less memory). Being able to adjust the batch size for ingesters and store-gateways independently means we'll be able to account for these differences - for example, given that a single series from a store-gateway is expected to be larger, we might want a single batch from a store-gateway to have fewer series.